src/Components/Popover.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react"
+import ReactDOM from "react-dom"
+import styled, { StyledFunction } from "styled-components"
+import { fadeIn, fadeOut } from "../Assets/Animations"
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  hide?: boolean
+  maxWidth?: string
+  adjustTop?: string
+  anchorRef?: HTMLElement
+  dismissOnClick?: boolean
+}
+
+interface PositionProps {
+  top: number
+  left: number
+  width: number
+}
+
+const Positionable: StyledFunction<
+  { position: PositionProps; ownWidth?: number } & Props &
+    React.HTMLProps<HTMLDivElement>
+> =
+  styled.div
+
+const Container = Positionable`
+  position: fixed;
+  top: ${props => props.adjustTop};
+  top: ${props => props.position && props.position.top}px;
+  left: ${props => props.position && props.position.left}px;
+  width: ${props => props.position && props.position.width};
+`
+
+const Tip = Positionable`
+  position: absolute;
+  display: ${props => (props.hide ? "none" : "block")};
+  visibility: ${props => (props.hide ? "hidden" : "visible")};
+  animation: ${props => (props.hide ? fadeOut : fadeIn)} 1s linear;
+  z-index: 1000;
+  // align-items: center;
+  // flex-direction: column;
+`
+
+const Arrow = Positionable`
+  width: 0;
+  height: 0;
+  border-bottom: 10px solid #000;
+  border-left: 10px solid transparent;
+  border-right: 10px solid transparent;
+  position: absolute;
+  left: ${props => props.position && props.position.width / 2 - 10}px;
+`
+// or:
+// width: 12px;
+// height: 12px;
+// transform: rotate(-45deg);
+// background: #000;
+
+const Message = Positionable`
+  ${/*props => (props.position && console.log(props.position.width, props.ownWidth)) || */ ""}
+  position: absolute;
+  top: 10px;
+  left: ${props =>
+    props.position && props.position.width / 2 - props.ownWidth / 2}px;
+  padding: 13px 15px;
+  background: #000;
+  color: #fff;
+  font-family: "Courier New", Courier, monospace;
+  cursor: default;
+  & span {
+    cursor: pointer;
+  }
+`
+
+export class Popover extends React.Component<Props, any> {
+  public messageRef
+
+  static defaultProps = {
+    hide: true,
+    maxWidth: "200px",
+    dismissOnClick: true,
+    // dismissTimeout: null
+  }
+
+  constructor(props) {
+    super(props)
+    const message = this.getMessage(props)
+    const hide = props.hide === true || !message
```

Comment:
For consistency sake, maybe 
```
const anchorPosition = this.getAnchorPosition(props.anchorRef),
```
which is then passed in similarly like the above two vars?